pkg/discovery/processor/cluster_processor.go

```diff
@@ -36,35 +37,46 @@ func NewClusterProcessor(kubeClient *cluster.ClusterScraper, kubeletClient *kube
 	return clusterProcessor
 }
 
+type ClusterValidationResult struct {
+	IsValidated      bool
+	UnreachableNodes []string
+}
+
 // Connects to the Kubernetes API Server and the nodes in the cluster.
-// If connection is successful a KubeCluster entity to represent the cluster is created
-func (processor *ClusterProcessor) ConnectCluster() (*repository.KubeCluster, error) {
+// ClusterProcessor is updated with the validation result.
```

Comment:
`ClusterProcessor` is also a global variable for `K8sDiscoveryClient`,  If it has state, make sure the update of the state is protected.